File: db/mssql.js
const sql = require('mssql');

async function runQuery(connectionConfig, query) {
  const { type, server, port, database, user, password, options } = connectionConfig;

  const config = {
    server,
    port,
    database,
    user,
    password,
    options: options || {},
  };

  const pool = await sql.connect(config);
  try {
    const result = await pool.request().query(query);
    return result.recordset;
  } finally {
    await pool.close();
  }
}

module.exports = { runQuery };

File: python-server/db/mssql.py
import pymssql

def run_query(conn_config, sql):
    connection = pymssql.connect(
        server=conn_config['server'],
        port=conn_config['port'],
        database=conn_config['database'],
        user=conn_config['user'],
        password=conn_config['password'],
    )
    try:
        cursor = connection.cursor(as_dict=True)
        cursor.execute(sql)
        return cursor.fetchall()
    finally:
        connection.close()
